pkg/kubefedctl/federate/util.go

```diff
@@ -187,3 +192,42 @@ func getResourcesInNamespace(config *rest.Config, namespace, skipAPIResourceName
 
 	return resourcesInNamespace, nil
 }
+
+// decodeUnstructuredFromFile reads a list of yamls into a slice of unstructured objects
+func decodeUnstructuredFromFile(filename string) ([]*unstructured.Unstructured, error) {
+	var f *os.File
+	if filename == "-" {
```

Comment:
This doesn't seem at all obvious to me.  Do other common commands employ this strategy to accept input from stdin?   And why wouldn't federate just assume stdin input when no args were supplied?